pkg/frontend/openshiftclusterkubeconfigcredentials_post_test.go

```diff
@@ -0,0 +1,337 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/api/v20210131preview"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/metrics/noop"
+	testdatabase "github.com/Azure/ARO-RP/test/database"
+)
+
+func TestPostOpenShiftClusterKubeConfigCredentials(t *testing.T) {
+	ctx := context.Background()
+
+	apis := map[string]*api.Version{
+		"2021-01-31-preview": api.APIs["2021-01-31-preview"],
+		"no-credentials": {
+			OpenShiftClusterConverter:       api.APIs["2021-01-31-preview"].OpenShiftClusterConverter,
+			OpenShiftClusterStaticValidator: api.APIs["2021-01-31-preview"].OpenShiftClusterStaticValidator,
+		},
+	}
+
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+
+	type test struct {
+		name           string
+		resourceID     string
+		apiVersion     string
+		fixture        func(*testdatabase.Fixture)
+		dbError        error
+		wantStatusCode int
+		wantResponse   func(*test) *v20210131preview.OpenShiftClusterAdminKubeconfig
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:       "cluster exists in db",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openShiftClusters/resourceName", mockSubID),
```

Comment:
`resourceID` is always the same in all test cases, isn't it? If so, I would just define single var out side the `test` struct.

Reply:
done :)